src/assets/Generator.Shared/LowLevelOperationHelpers.cs

```diff
@@ -62,5 +67,172 @@ public static Operation<Pageable<BinaryData>> ProcessMessage(HttpPipeline pipeli
             }
             return operation;
         }
+
+        private class ConvertOperation<TFrom, TTo> : Operation<TTo>
+            where TFrom : notnull
+            where TTo : notnull
+        {
+            private readonly Operation<TFrom> _operation;
+            private readonly ClientDiagnostics _diagnostics;
+            private readonly string _operationName;
+            private readonly string _updateStatusScopeName;
+            private readonly Func<Response, TTo> _convertFunc;
+            private Response<TTo>? _response;
+
+            public ConvertOperation(Operation<TFrom> operation, ClientDiagnostics diagnostics, string operationName, Func<Response, TTo> convertFunc)
+            {
+                _operation = operation;
```

Comment:
Any parameter check needed in the constructor?

Reply:
No. This is an internal class.